src/cache/base.py

```diff
@@ -0,0 +1,102 @@
+from functools import wraps
+from typing import Any, Callable
+from sys import getsizeof
+import pickle
+import time
+
+
+class BaseCache:
+    def __init__(self) -> None:
+        # these specific elements can be change subClasses
+        self.dumper = pickle.dumps
+        self.loader = pickle.loads
+
+    def _log_cache_failed(self, key: str, logger: Any) -> None:
+        """Just some logging"""
+        logger.debug(f">> cache failed on custom_condition for {key=}")
+
+    def _log_cache_hit(self, key: str, logger: Any) -> None:
+        """Just some logging"""
+        logger.debug(f">> cache hit for {key=}")
+
+    def _log_cache_missed(self, key: str, logger: Any) -> None:
+        """Just some logging"""
+        logger.debug(f">> cache missed for {key=}")
+
+    def _size_of(self, value: Any) -> int:
+        """To return the memory size of a value"""
+        return getsizeof(value)
+
+    def _build_key(self, *args: Any, **kwargs: Any) -> str:
+        """The key builder for our cache system"""
+        return f"{args}-{kwargs}"
+
+    def _build_value(self, value: Any) -> bytes:
+        """
+        We need to have an uniform way to build the value and how it will be stored
+        as bytes for perfs requirements
+        """
+        return self.dumper(
+            {
+                "value": value,
+                "at": time.time(),
+                "size_of": self._size_of(value),
+            }
+        )
```

Comment:
This is not OK for file cache, see comment blow.
Maybe we can rely on the file modification time and the size provided by the file system?

Reply:
i made it (_build_value) as an abstract method, an it is rewrite in the `FileCache` class !